app/javascript/components/ExpressDonation/ExpressDonation.js

```diff
@@ -73,15 +87,28 @@ export class ExpressDonation extends Component<Props, State> {
   }
 
   async onFailure(reason: any): any {
-    this.setState({ submitting: false });
+    reason.responseJSON && reason.responseJSON.immediate_redonation
+      ? this.setState({
+          optForRedonation: reason.responseJSON.immediate_redonation,
+        })
+      : null;
```

Comment:
So the main issue I have with this `onFailure` function is that we're setting state twice for no reason. Let's merge both into one.